kafka-rest/src/test/java/io/confluent/kafkarest/integration/AbstractConsumerTest.java

```diff
@@ -206,33 +220,46 @@ protected String startConsumeMessages(
       GenericType<List<RecordType>> responseEntityType,
       Converter converter,
       Function<RecordType, ConsumerRecord<ClientK, ClientV>> fromJsonWrapper) {
-    Response response = request(instanceUri + "/records").accept(accept).get();
-    assertOKResponse(response, responseMediatype);
-    List<RecordType> consumed = TestUtils.tryReadEntityOrLog(response, responseEntityType);
-    try {
-      assertEquals(records.size(), consumed.size());
-    } catch (AssertionError e) {
-      pause();
-      Response response2 = request(instanceUri + "/records").accept(accept).get();
-      assertOKResponse(response2, responseMediatype);
-      List<RecordType> consumed2 = TestUtils.tryReadEntityOrLog(response2, responseEntityType);
-      consumed.addAll(consumed2);
+
+    int retryCount = 0;
+    final int TOTAL_RETRIES = 3;
+    List<RecordType> consumed = new ArrayList<>();
+
+    while (retryCount < TOTAL_RETRIES) {
+      Response response = request(instanceUri + "/records").accept(accept).get();
+      assertOKResponse(response, responseMediatype);
+      consumed.addAll(TestUtils.tryReadEntityOrLog(response, responseEntityType));
```

Comment:
I think that `testWithRetry` should work OK with externally kept `final` collection being modified - is there a different reason to prefer a manual retry approach?